patterns.js

```diff
@@ -32,6 +32,7 @@ var RO = "^RO[0-9]{2,10}$";
 var SE = "^SE[0-9]{10,10}01$";
 var SI = "^SI[0-9]{8,8}$";
 var SK = "^SK[0-9]{10,10}$";
+var CH = "^CHE[0-9]{9,9}$";
```

Comment:
Just checking out the wikipedia article on non-EU VAT codes and I found the following about Swiss VAT codes:

> 6 digits (up to 31/12/2013). CHE 9 numeric digits plus TVA/MWST/IVA e.g. CHE-123.456.788 TVA[20] The last digit is a MOD11 checksum digit build with weighting pattern: 5,4,3,2,7,6,5,4[21]

The format you've supplied for the tests below, and the regex here, don't include the intermediate dots. Is excluding the dots common in Switzerland when entering a VAT code, or are they often seen with the dots? I also noticed the missing hyphen as well.

I'd be incline to prefer sticking to the Swiss standards of using the hyphen and dots but if this isn't actually used by anyone then the pragmatic approach would be as you've got here.

Do you have any thoughts? Being the one using Swiss VAT codes, I'm assuming you might have some better insight than me!

Reply:
Hi @BlueHatbRit, first of all thank you for the fast response! 👍

You are right, by the standards all of the variations are allowed:

`CHE112947556,
  CHE 112947556,
  CHE 112.947.556,
  CHE-112.947.556 IVA,
  CHE112.947.556TVA,
  CHE 112947556 MWST`

IMHO the first option is the usual one, but you never know from where will someone copy the VAT number and enter it in a different format. If we want to do it right all the options should be supported. That is why I prepared new regex which covers all the cases (at least all that I could think of :) ). I also added some more test cases. Take a look and let me know what you think!